src/index.ts

```diff
@@ -5,7 +5,39 @@
  * @author Kent C. Dodds <me@kentcdodds.com> (https://kentcdodds.com)
  */
 import removeAccents from 'remove-accents'
+import type {Merge} from 'type-fest'
 
+type KeyAttributes = {
+  threshold?: number
+  maxRanking: number
+  minRanking: number
+}
+type RankingInfo = {
+  rankedValue: string
+  rank: number
+  keyIndex: number
+  keyThreshold: number | undefined
+}
+
+declare function valueGetterKey<ItemType>(item: ItemType): string
```

Comment:
for the declare function, I'd also prefer going with an interface. Unfortunately, this will cause a couple of other types you're exporting to be generics (but it's not terrible);

```ts
interface ValueGetterKey<ItemType> {
  (item: ItemType): string
}

interface BaseSorter<ItemType> {
  (a: RankedItem<ItemType>, b: RankedItem<ItemType>): number
}

interface KeyAttributesOptions<ItemType> {
  key?: string | ValueGetterKey<ItemType>
  threshold?: number
  maxRanking?: number
  minRanking?: number
}

type KeyOption<ItemType> =
  | KeyAttributesOptions<ItemType>
  | ValueGetterKey<ItemType>
  | string

// ItemType = unknown allowed me to make these changes without the need to change the current tests
interface MatchSorterOptions<ItemType = unknown> {
  keys?: Array<KeyOption<ItemType>>
  threshold?: number
  baseSort?: BaseSorter<ItemType>
  keepDiacritics?: boolean
}
// ...
function matchSorter<ItemType>(
  items: Array<ItemType>,
  value: string,
  options: MatchSorterOptions<ItemType> = {}, // I make sure to pass `ItemType` to the type I defaulted to unknown to get proper type inference
): Array<ItemType> {
```